src/__tests__/index.ts

```diff
@@ -189,6 +203,21 @@ test(
   30000
 )
 
+test("Functions passed into non-source functions remain equal", () => {
+  const code = `
+  function t(x, y, z) {
+    return x + y + z;
+  }
+  head(pair(t, t)) === t && t(1, 2, 3) === 6;
```

Comment:
I think to be more robust against future library changes (what if head/pair are later implemented in pure Source), you could use `defineSymbol` from `createContext.ts` to create your own built-in that's guaranteed to be non-source.

Reply:
Yes, it was probably wrong to rely on external "builtins".